hibernate-core/src/main/java/org/hibernate/tool/schema/internal/AbstractSchemaMigrator.java

```diff
@@ -139,6 +135,8 @@ public void doMigration(Metadata metadata, ExecutionOptions options, TargetDescr
 		}
 	}
 
+	protected abstract DatabaseInformation getDatabaseInformation(DdlTransactionIsolator ddlTransactionIsolator, Namespace namespace);
```

Comment:
Hibernate strives to maintain backward compatibility so new method is supposed to be introduced via JDK8's default interface method mechanism (our client might be compiling based on the public methods we have provided and this new abstract method might break all their existing code).

Reply:
it is an internal so it is not a problem

Reply:
I will update the pull request, and make this method none-abstract, which will implement the "current" behavior. "				final DatabaseInformation databaseInformation = Helper.buildDatabaseInformation(
						tool.getServiceRegistry(),
						ddlTransactionIsolator,
						metadata.getDatabase().getDefaultNamespace().getName()
				);"

and I will overwrite this method in the "GroupedSchemaMigratorImpl"

As a result the implementation of "IndividuallySchemaMigratorImpl" and "CheckForExistingForeignKeyTest" are no longer affected by this pull request.

Reply:
> it is an internal so it is not a problem

Thanks. I forgot it is in 'internal' package.